ballista/scheduler/src/cluster/mod.rs

```diff
@@ -1020,14 +1024,10 @@ mod test {
             specification: ExecutorSpecification { task_slots: 32 },
         };
 
-        if let Some(task) = graph.pop_next_task(&executor.id)? {
-            let task_status = mock_completed_task(task, &executor.id);
-            graph.update_task_status(&executor, vec![task_status], 1, 1)?;
-        }
-
-        graph.revive();
+        // complete first stage
+        revive_graph_and_complete_next_stage(&mut graph)?;
```

Comment:
this helper method just finds the next stage that has an available task and completes the stage. This was required because we create scans or memtables with 2 partitions, which result in 2 tasks in the 1st stage. [This](https://github.com/apache/datafusion/issues/12611) only causes regression when the input doesn't have multiple partitions